### PR TITLE
FS-2014: Add endpoint to retrieve banner state

### PR DESCRIPTION
**Pull request by tferns**
[Add endpoint to retrieve banner state](https://github.com/communitiesuk/funding-service-design-assessment-store/pull/75/commits/8279607efe5281a364be1466a47740d429ba48d4)
- Update metadata to use Enum as was seeing error

[Add flag routes and database model/migration](https://github.com/communitiesuk/funding-service-design-assessment-store/pull/75/commits/2f1a616e6145b90c15dc4ad863ebfa58d967f992)
- Get request to return all flags for a given application
- Post request to create a flag for a given application
- Migration script introduced to create the flags table in db

### CODE REVIEW

File: db/models/flags/flags.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+    section_to_flag = db.Column("section_to_flag", db.Text(), nullable=False)
+
+    flag_type = db.Column("flag_type", ENUM(FlagType), nullable=False)
+
+    resolution_reason = db.Column(
+        "resolution_reason", ENUM(ResolutionType), nullable=True
+    )
+
+    application_id = db.Column(
```
Comment (Adam-W1) [inline db/models/flags/flags.py:35]: just wondering if there is a restraint that there should only ever be one flag per application at a time? Unsure on the business rules around this so more of a Q tbf

Comment (tferns) [reply db/models/flags/flags.py:35]: Hey, good point!  From what I could gather, there _might_ be different types of flagging in the future... so maybe something for the future?

File: openapi/api.yml [hunk 2]
```diff
@@ -155,6 +157,67 @@ paths:
...
+      responses:
+        201:
+          description: SUCCESS - Flag for Application Submitted
+          content:
+            application/json:
+              schema:
+                type: object
+                items:
+                  $ref: 'components.yml#/components/schemas/Flag'
+        401:
```
Comment (adamdavies1) [inline openapi/api.yml:210]: Is this Supposed to ve "Could not Save Flag"?

**Review by adamdavies1 — APPROVED** [review]
LGTM

File: openapi/api.yml (hunk 2 above)
Comment (tferns) [reply openapi/api.yml:210]: Good eye, have fixed in https://github.com/communitiesuk/funding-service-design-assessment-store/pull/75/commits/b6b34c0ed88dcf65c191572addf4a181fa67a8c3

**Pull request merged**